app.rb

```diff
@@ -2,29 +2,46 @@
 require 'bundler/setup'
 require_relative 'publisher.rb'
 require_relative 'subscriber.rb'
+require_relative 'q_it_errors'
 
-def run_service(options)
+SERVICES = ['publish', 'subscribe']
+
+def run_service(service, options)
+  if valid_options?(options)
+    service = service
+    case service
+    when 'publish'
+      publisher = Publisher.new(options)
+      publisher.start
+    when 'subscribe'
+      subscriber = Subscriber.new(options)
+      subscriber.start
+    else
+      raise QItNoServiceError, "Invalid service name. Try 'publish' or 'subscribe'." if (service.nil? || !SERVICES.include?(service.downcase))
```

Comment:
Use `exclude?` instead of `!include?`.

Reply:
`exclude?` is not pure ruby. Does not work here.